src/hotspot/share/gc/shared/gc_globals.hpp

```diff
@@ -212,12 +212,12 @@
   /* where does the range max value of (max_jint - 1) come from? */         \
   product(size_t, MarkStackSizeMax, NOT_LP64(4*M) LP64_ONLY(512*M),         \
           "Maximum size of marking stack")                                  \
-          range(1, (max_jint - 1))                                          \
+          range(1, (SIZE_MAX - 1))                                          \
```

Comment:
This is a large increase in the max size.  Not at all sure it's appropriate.  Similarly for MarkStackSize below.